cpp/dolfinx/fem/discreteoperators.h

```diff
@@ -306,15 +312,20 @@ void interpolation_matrix(const fem::FunctionSpace& V0,
       auto _U = xt::view(mapped_values, p, xt::all(), xt::all());
       pull_back_fn1(_U, _u, _K, 1.0 / detJ[p], _J);
     }
+
     // Apply interpolation matrix to basis values of V0 at the interpolation
     // points of V1
-    for (std::size_t i = 0; i < mapped_values.shape(1); ++i)
+    if (interpolation_ident)
+      _A.assign(xt::transpose(mapped_values, {0, 2, 1}));
+    else
     {
-      auto _mapped_values = xt::view(mapped_values, xt::all(), i, xt::all());
-      dolfinx::fem::impl::interpolation_apply(Pi_1, _mapped_values, local1,
-                                              bs1);
-      for (std::size_t j = 0; j < local1.size(); j++)
-        A[element0->space_dimension() * j + i] = local1[j];
+      for (std::size_t i = 0; i < mapped_values.shape(1); ++i)
+      {
+        auto _mapped_values = xt::view(mapped_values, xt::all(), i, xt::all());
+        impl::interpolation_apply(Pi_1, _mapped_values, local1, bs1);
+        for (std::size_t j = 0; j < local1.size(); j++)
+          A[element0->space_dimension() * j + i] = local1[j];
```

Comment:
Should `element0->space_dimension()` be taken out of the loop? Does if affect performance?

Reply:
I changed it to 
```c++
A[space_dim0 * j + i] = = local1[j];
```
and there is a minor change in performancen for low order elements.